src/cargo/util/toml/mod.rs

```diff
@@ -1944,13 +1939,18 @@ fn detailed_dep_to_dependency<P: ResolveToPath + Clone>(
     }
 
     if let Some(p) = orig.public {
-        cx.features.require(Feature::public_dependency())?;
-
         if dep.kind() != DepKind::Normal {
             bail!("'public' specifier can only be used on regular dependencies, not {:?} dependencies", dep.kind());
         }
 
-        dep.set_public(p);
+        if !cx.config.cli_unstable().public_dependency {
+            cx.warnings.push(format!(
+                "{name} uses 'public' specifier, pass `-Zpublic-dependency` to enable support for it",
+                name = dep.name_in_toml(),
```

Comment:
Is nessary to set the dep.public as Nonek or false ?

Reply:
Leaving `dep.set_public` alone is likely the right call as that is what was done before without this unstable feature.